### PR TITLE
graphql: return correct, non-empty language for repositories

### PR DESCRIPTION
**Pull request by beyang**
Related issue: https://github.com/sourcegraph/sourcegraph/issues/2586

**Post-merge TODO:**
- [ ] Cherry-pick this onto 3.4 and 3.5.
- [ ] Cut new 3.4 and 3.5 releases for $CUSTOMER.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/repository.go [hunk 1]
```diff
@@ -145,8 +143,25 @@ func (r *repositoryResolver) DefaultBranch(ctx context.Context) (*gitRefResolver
...
+	// Note: the repository database field is no longer updated as of
+	// https://github.com/sourcegraph/sourcegraph/issues/2586, so we do not use it anymore and
+	// instead compute the language on the fly.
+
+	commitID, err := backend.Repos.ResolveRev(ctx, r.repo, "")
+	if err != nil {
+		return ""
+	}
+
+	inventory, err := backend.Repos.GetInventory(ctx, r.repo, commitID)
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/repository.go]: does this hit a cache?

File: cmd/frontend/types/types.go [hunk 2]
```diff
@@ -37,8 +37,13 @@ type Repo struct {
 
 	// Description is a brief description of the repository.
 	Description string
+
+	// DEPRECATED: this field is always empty for new repositories as of
```
Comment (keegancsmith) [inline cmd/frontend/types/types.go]: just remove the field, we don't expose it anywhere else.

File: cmd/frontend/graphqlbackend/repository.go (hunk 1 above)
Comment (beyang) [reply cmd/frontend/graphqlbackend/repository.go]: Yes: https://github.com/sourcegraph/sourcegraph/blob/6ec1acc53873f610c3eb036865ae4c7e903ed8b5//cmd/frontend/backend/repos.go#L173-L188

File: cmd/frontend/types/types.go (hunk 2 above)
Comment (beyang) [reply cmd/frontend/types/types.go]: Will remove in a follow-up PR

**Pull request merged**